core/server/data/migrations/versions/4.43/2022-04-01-10-13-add-post-newsletter-relation.js

```diff
@@ -0,0 +1,112 @@
+const logging = require('@tryghost/logging');
+const DatabaseInfo = require('@tryghost/database-info');
+const commands = require('../../../schema/commands');
+
+const table = 'posts';
+const column = 'newsletter_id';
+const targetTable = 'newsletters';
+const targetColumn = 'id';
+
+const columnDefinition = {
+    type: 'string',
+    maxlength: 24,
+    nullable: true,
+    references: `${targetTable}.${targetColumn}`
+};
+
+/**
+ * This migration is adding a new column `newsletter_id` to the table posts
+ * that is a foreign key to `newsletters.id`.
+ *
+ * It isn't using the existing utils because of a performance issue. In MySQL,
+ * adding a new row without `algorithm=copy` uses the INPLACE algorithm which
+ * was too slow on big `posts` tables (~3 minutes for 10k posts). Switching to
+ * the COPY algorithm fixed the issue (~3 seconds for 10k posts).
+ */
+module.exports = {
```

Comment:
Can we refactor to use the util for transactional migrations? Like https://github.com/TryGhost/Ghost/blob/caea330647b70c300428764de52a05c873ee5652/core/server/data/migrations/versions/4.3/10-populate-stripe-price-id-in-subscriptions.js#L4

Reply:
done ✅